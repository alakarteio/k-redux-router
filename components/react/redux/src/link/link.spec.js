/* eslint-disable
  react/jsx-filename-extension,
  function-paren-newline,
*/
/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import Link from './link'

describe('Link', () => {
  it('should print the complete link -with path params and query params-', () => {
    const compiled = jest.fn(() => '/full/path/compiled')
    const rendered = renderer.create(
      <Link
        href={{
          base: '/path/:with/some/:params',
          compiled,
        }}
        with="with-value"
        params="params-value"
        query={{
          this: 'is',
          two: 'query-params',
        }}
      />,
    )

    expect({
      rendered,
      compiled: compiled.mock.calls,
    }).toMatchSnapshot()
  })
})
