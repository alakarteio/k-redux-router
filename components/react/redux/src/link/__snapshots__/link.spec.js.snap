// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should print the complete link -with path params and query params- 1`] = `
Object {
  "compiled": Array [
    Array [
      Object {
        "children": undefined,
        "className": undefined,
        "params": "params-value",
        "query": Object {
          "this": "is",
          "two": "query-params",
        },
        "with": "with-value",
      },
    ],
  ],
  "rendered": <a
    className={undefined}
    href="/full/path/compiled?this=is&two=query-params"
    onClick={undefined}
  />,
}
`;
